### PR TITLE
docs(website): add Discord and Open Collective logo links

### LINKED ISSUE #8627: Docs: Put Discord and/or donation link(s) in header
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

@bradzacher and I were chatting on a Twitch stream yesterday, and noted that our header doesn't have a link to [our Discord](https://discord.com/invite/FSxKq8Tdyg). We should add one next to the GitHub link.

You can find Discord's branding assets here: https://discord.com/branding

### Affected URL(s)

_all of them_

### PR DESCRIPTION
**Pull request by abrahamguo**
## PR Checklist

- [x] Addresses an existing open issue: fixes #8627
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview
Per @JoshuaKGoldberg, adds:
- Discord: to the header
-- Should we add all socials from the footer (Mastadon, Stack Overflow, Twitter)?
-- Should we add some sort of tooltip on hover?
- Open Collective: to the footer > More
-- Should it be `Become a financial sponsor`, matching the homepage link, or should it just be `OpenCollective`, matching the other links in the footer?
-- Where can I get the SVG logo from, for OC?

### CODE REVIEW

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
✨ cool!

**Comment by armano2** [comment]
open collective logo, extracted only icon from https://opencollective.com/static/images/opencollectivelogo.svg
```
<svg xmlns="http://www.w3.org/2000/svg" width="38" height="38"><path fill="#7FADF2" d="M34.62 8.179C36.75 11.249 38 14.979 38 19s-1.25 7.75-3.38 10.821L29.697 24.9A12.16 12.16 0 0 0 31.214 19c0-2.14-.55-4.15-1.516-5.9l4.921-4.921ZM29.82 3.38 24.9 8.302A12.16 12.16 0 0 0 19 6.786C12.254 6.786 6.786 12.254 6.786 19S12.254 31.214 19 31.214c2.14 0 4.15-.55 5.9-1.516l4.921 4.921A18.912 18.912 0 0 1 19 38C8.507 38 0 29.493 0 19S8.507 0 19 0c4.021 0 7.75 1.25 10.821 3.38Z"/><path fill="#B8D3F4" d="M34.62 8.179C36.75 11.249 38 14.979 38 19s-1.25 7.75-3.38 10.821L29.697 24.9A12.16 12.16 0 0 0 31.214 19c0-2.14-.55-4.15-1.516-5.9l4.921-4.921Z"/></svg>
```

File: packages/website/docusaurus.config.mts [hunk 1]
```diff
@@ -154,6 +160,11 @@ const themeConfig: AlgoliaThemeConfig & ThemeCommonConfig = {
             href: `${githubUrl}/issues/new/choose`,
             className: 'bug-report-link image-link social-link-icon',
           },
+          {
+            label: 'Become a financial sponsor',
```
[the remark this answers is not in the document]
Comment (armano2) [reply packages/website/docusaurus.config.mts]: open collective logo, extracted only icon from https://opencollective.com/static/images/opencollectivelogo.svg
```
<svg xmlns="http://www.w3.org/2000/svg" width="38" height="38">
  <path fill="#7FADF2" d="M34.62 8.179C36.75 11.249 38 14.979 38 19s-1.25 7.75-3.38 10.821L29.697 24.9A12.16 12.16 0 0 0 31.214 19c0-2.14-.55-4.15-1.516-5.9l4.921-4.921ZM29.82 3.38 24.9 8.302A12.16 12.16 0 0 0 19 6.786C12.254 6.786 6.786 12.254 6.786 19S12.254 31.214 19 31.214c2.14 0 4.15-.55 5.9-1.516l4.921 4.921A18.912 18.912 0 0 1 19 38C8.507 38 0 29.493 0 19S8.507 0 19 0c4.021 0 7.75 1.25 10.821 3.38Z"/>
  <path fill="#B8D3F4" d="M34.62 8.179C36.75 11.249 38 14.979 38 19s-1.25 7.75-3.38 10.821L29.697 24.9A12.16 12.16 0 0 0 31.214 19c0-2.14-.55-4.15-1.516-5.9l4.921-4.921Z"/>
</svg>
```

**Comment by armano2** [comment]
open collective logo, extracted only icon from https://opencollective.com/static/images/opencollectivelogo.svg
```
<svg xmlns="http://www.w3.org/2000/svg" width="38" height="38">
  <path fill="#7FADF2" d="M34.62 8.179C36.75 11.249 38 14.979 38 19s-1.25 7.75-3.38 10.821L29.697 24.9A12.16 12.16 0 0 0 31.214 19c0-2.14-.55-4.15-1.516-5.9l4.921-4.921ZM29.82 3.38 24.9 8.302A12.16 12.16 0 0 0 19 6.786C12.254 6.786 6.786 12.254 6.786 19S12.254 31.214 19 31.214c2.14 0 4.15-.55 5.9-1.516l4.921 4.921A18.912 18.912 0 0 1 19 38C8.507 38 0 29.493 0 19S8.507 0 19 0c4.021 0 7.75 1.25 10.821 3.38Z"/>
  <path fill="#B8D3F4" d="M34.62 8.179C36.75 11.249 38 14.979 38 19s-1.25 7.75-3.38 10.821L29.697 24.9A12.16 12.16 0 0 0 31.214 19c0-2.14-.55-4.15-1.516-5.9l4.921-4.921Z"/>
</svg>
```

https://docs.opencollective.com/help/about/introduction#media-logo

File: packages/website/docusaurus.config.mts [hunk 2]
```diff
@@ -154,6 +160,11 @@ const themeConfig: AlgoliaThemeConfig & ThemeCommonConfig = {
             href: `${githubUrl}/issues/new/choose`,
             className: 'bug-report-link image-link social-link-icon',
           },
+          {
+            label: 'OpenCollective',
```
Comment (JoshuaKGoldberg) [inline packages/website/docusaurus.config.mts:164]: Typo:

```suggestion
            label: 'Open Collective',
```

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM, thanks! ✨

**Pull request merged**